### PR TITLE
Strava Provider Implementation

### PR DESCRIPTION
**Pull request by segilbert**
Please consider this pull request contribution for Strava OAuth Provider support to your fantastic project.  Let me know.

I also implemented and outlined how to leverage OwinOAuthProviderConfig.cs to help prevent committing clientId and clientSecret data into version control.

Cheers,
Sean

### CODE REVIEW

File: OwinOAuthProvidersDemo/Web.config [hunk 1]
```diff
@@ -9,13 +9,18 @@
     <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
   <!-- For more information on Entity Framework configuration, visit http://go.microsoft.com/fwlink/?LinkID=237468 --></configSections>
   <connectionStrings>
-    <add name="DefaultConnection" connectionString="Data Source=(LocalDb)\v11.0;AttachDbFilename=|DataDirectory|\aspnet-OwinOAuthProvidersDemo-20131113093838.mdf;Initial Catalog=aspnet-OwinOAuthProvidersDemo-20131113093838;Integrated Security=True" providerName="System.Data.SqlClient" />
```
Comment (kappa7194) [inline OwinOAuthProvidersDemo/Web.config:12]: Please undo this change. You replaced the standard connection string for SQLLocalDB with a connection valid only for you.

File: Owin.Security.Providers/Strava/Provider/IStravaAuthenticationProvider.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Threading.Tasks;
+
+namespace Owin.Security.Providers.Strava.Provider
+{
+    public interface IStravaAuthenticationProvider
+    {
+        /// <summary>
+        /// Invoked whenever LinkedIn succesfully authenticates a user
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/Provider/IStravaAuthenticationProvider.cs:8]: You left "LinkedIn" in the XML documentation.

File: Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+using Microsoft.Owin.Security;
+using Owin.Security.Providers.Strava.Provider;
+
+namespace Owin.Security.Providers.Strava
+{
+    public class StravaAuthenticationOptions : AuthenticationOptions
+    {
+        /// <summary>
+        ///     Gets or sets the a pinned certificate validator to use to validate the endpoints used
+        ///     in back channel communications belong to LinkedIn.
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs:14]: You left "LinkedIn" in the XML documentation.

File: Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        ///     The pinned certificate validator.
+        /// </value>
+        /// <remarks>
+        ///     If this property is null then the default certificate checks are performed,
+        ///     validating the subject name and if the signing chain is a trusted party.
+        /// </remarks>
+        public ICertificateValidator BackchannelCertificateValidator { get; set; }
+
+        /// <summary>
+        ///     The HttpMessageHandler used to communicate with LinkedIn.
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs:26]: You left "LinkedIn" in the XML documentation.

File: Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        ///     This cannot be set at the same time as BackchannelCertificateValidator unless the value
+        ///     can be downcast to a WebRequestHandler.
+        /// </summary>
+        public HttpMessageHandler BackchannelHttpHandler { get; set; }
+
+        /// <summary>
+        ///     Gets or sets timeout value in milliseconds for back channel communications with LinkedIn.
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs:33]: You left "LinkedIn" in the XML documentation.

File: Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        /// </summary>
+        /// <value>
+        ///     The back channel timeout in milliseconds.
+        /// </value>
+        public TimeSpan BackchannelTimeout { get; set; }
+        /// <summary>
+        ///     The request path within the application's base path where the user-agent will be returned.
+        ///     The middleware will process this request when it arrives.
+        ///     Default value is "/signin-linkedin".
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs:42]: You left the default path for LinkedIn in the XML documentation.

File: Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        ///     Get or sets the text that the user can display on a sign in user interface.
+        /// </summary>
+        public string Caption
+        {
+            get { return Description.Caption; }
+            set { Description.Caption = value; }
+        }
+
+        /// <summary>
+        ///     Gets or sets the LinkedIn supplied Client ID
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs:56]: You left "LinkedIn" in the XML documentation.

File: Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        /// </summary>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the LinkedIn supplied Client Secret
```
Comment (kappa7194) [inline Owin.Security.Providers/Strava/StravaAuthenticationOptions.cs:61]: You left "LinkedIn" in the XML documentation.

**Comment by kappa7194** [comment]
I think splitting the provider and the secrets storing method change in two different pull requests dould be better, as they can be evaluated separately.

File: OwinOAuthProvidersDemo/App_Start/Startup.Auth.cs [hunk 9]
```diff
@@ -51,203 +19,11 @@ public void ConfigureAuth(IAppBuilder app)
             });
             // Use a cookie to temporarily store information about a user logging in with a third party login provider
             app.UseExternalSignInCookie(DefaultAuthenticationTypes.ExternalCookie);
-           //app.UseDeviantArtAuthentication("id", "secret");
```
Comment (kappa7194) [inline OwinOAuthProvidersDemo/App_Start/Startup.Auth.cs:54]: Why did you remove every usage example besides your implementation?

**Comment by jerriep** [comment]
Thank you @segilbert 

I will look at all PRs this coming Friday

**Comment by jerriep** [comment]
This PR is not valid anymore...

**Comment by jerriep** [comment]
@segilbert We are looking for help moving to ASP.NET 5. Please read http://www.jerriepelser.com/blog/calling-contributors-for-oauth-providers
